### ISSUE TITLE
Does not build on distros that provide libclangTooling as libclang-cpp.so

### ISSUE THREAD

**Issue by vimpostor** [KKoovalsky/Tsepepe#8]
Some modern distros build clang such that there are no individual libraries libclangTooling.so etc, but only a single libclang-cpp library. For example Arch Linux [does so since clang 10](https://bugs.archlinux.org/task/66283#comment188425).

It would be cool if there was an option to enable Tsepepe to build on those distros (right now the build fails, because Tsepepe hardcodes linking to `clangTooling` everywhere).

For example qtcreator added an option to allow distros to opt-in to linking to libclang-cpp instead via an `CLANGTOOLING_LINK_CLANG_DYLIB` cmake option like this: https://github.com/qt-creator/qt-creator/blob/ee98c98676d96bbd633a92c976077adc129b82a1/cmake/FindClang.cmake#L7

It would be nice, if something similar could be added to this project. Would you accept a patch adding such an option?

P.S.: FYI the clang 15 build currently fails because of a missing header include:
```diff
diff --git a/src/src/libclang_utils/misc_utils.cpp b/src/src/libclang_utils/misc_utils.cpp
index d8ae744..548f5a2 100644
--- a/src/src/libclang_utils/misc_utils.cpp
+++ b/src/src/libclang_utils/misc_utils.cpp
@@ -9,6 +9,7 @@
 #include <iostream>
 #include <regex>
 
+#include <clang/Basic/SourceManager.h>
 #include <clang/Lex/Lexer.h>
 
 using namespace clang;
```

**Comment by KKoovalsky** [comment]
Hello! You are more than welcome to create such a patch. If more distros could be supported, then I am all for it.

Ad. P.S. I have pushed a commit with the fix you have proposed, but I still can't compile it with `clang 15.0.6` due to problems with connecting `std::filesystem` iterators with the `std::ranges::view`. Now I can see that the include paths are taken from the system tree, instead of the system headers where `clang` is located (I use `export CC=/some/local/user/path`, which is not a system-wide visible compiler path). I need to check it out with a toolchain file.

**Comment by vimpostor** [comment]
> Hello! You are more than welcome to create such a patch.

Thanks, I have pushed the fixes in #9 :)

> I have pushed a commit with the fix you have proposed, but I still can't compile it with `clang 15.0.6` due to problems with connecting `std::filesystem` iterators with the `std::ranges::view`

I didn't formulate that well... I didn't mean compiling with clang 15, I meant compiling with gcc but linking against libclang/libllvm 15, indeed the clang build doesn't work right now due to missing `<ranges>` support.
